spec.html

```diff
@@ -317,9 +324,14 @@ <h3>Notes</h3>
   <li>Generic binary subtype - This is the most commonly used binary subtype and should be the 'default' for drivers and tools.</li>
   <li>The BSON "binary" or "BinData" datatype is used to represent arrays of bytes. It is somewhat analogous to the Java notion of a ByteArray. BSON binary values have a <em>subtype</em>. This is used to indicate what kind of data is in the byte array. Subtypes from zero to 127 are predefined or reserved. Subtypes from 128-255 are user-defined.
```

Comment:
This is kinda problematic. This describes the BinData subtype as unsigned. Based on a quick look, I think this matches the implementation. For example, in the shell:

```
a = BinData(-3, "gf1UcxdHTJ2HQ/EGQrO7mQ==")
Error: BinData subtype must be a Number between 0 and 255 inclusive :
```

I think the server code interprets these bindata subtypes as unsigned as well. So it seems we need to define a notation like `unsigned_type(n)` and use it to document the BinData subtypes.

Separately, I noticed that this page does not document subtype 8. See https://github.com/10gen/mongo/blob/64d32d7fff7e8c8956c1478d4afc0c09f3abd669/src/mongo/bson/bsontypes.h#L208. Perhaps adding that should be pursued as a separate improvement?

Reply:
FYI #79

Reply:
Got it, thanks for the link to #79! Resolving.